src/ansys/units/quantity.py

```diff
@@ -5,8 +5,15 @@
 
 import ansys.units as ansunits
 
+try:
+    import numpy as np
 
-class Quantity(float):
+    _array = np.array
```

Comment:
It's just `np` that you should be caching here: although you do checks in terms of `_array`, you are not actually using it but instead reaccessing `array` through `np`. You are also accessing `ndarray`.